docs/03-code-internals/23-transformers.md

```diff
@@ -0,0 +1,112 @@
+---
+title: Using Transformers to customize client-side values and behaviour
+short_title: Transformers
+id: client-side-transformers
+---
+ Discourse core includes a number of "transformer" hooks which can be used to customize client behaviour. These fall into two categories:
+
+- **Value Transformers** take an output from Discourse core, and optionally modify it, or replace it with something else
+
+- **Behavior Transformers** can replace a particular piece of logic in Discourse core, and optionally call the original implementation.
+
+Each transformer hook is referenced by a name, and multiple plugins/themes can register transformers against the same hook.
+
+## Value Transformers
+
+Value transformers can be registered using  `api.registerValueTransformer`:
+
+> ### `registerValueTransformer` => `boolean`
+> 
+> | Param | Type | Description |
+> |--|--|--|
+> | `transformerName` | `string` | the name of the transformer |
+> | `valueCallback` | `function({value, context})` | callback to be used to transform the value |
+
+The `valueCallback` will be passed two named arguments: the original value, and a context object. It should return the transformed value. Or, if no transformation is required, it should return the original value.
+
+Callbacks can access the provided context for the current transformer hook, or they can reference other data sources. Many value transformers are executed in autotracking contexts, which means that referencing reactive state will cause the transformer to be automatically re-evaluated when that state changes.
+
+For example, to modify the link URL of the logo on mobile devices:
+
+```js
+api.registerValueTransformer("home-logo-href", ({ value, context }) => {
+  const site = api.lookup("service:site");
+  if (site.mobileView) {
+    return "/latest";
+  } else {
+    return value;
+  }
+});
+```
+
+## Behavior Transformers
+
+Behavior transformers are similar, but instead of receiving and returning a value, they wrap some core behaviour, and can optionally call the core implementation.
+
+> ### `registerBehaviorTransformer` => `boolean`
+> 
+> | Param | Type | Description |
+> |--|--|--|
+> | `transformerName` | `string` | the name of the transformer |
+> | `behaviorCallback` | `function({next, context})` | callback to be used to transform or override the behavior |
+
+The `behaviorCallback` is passed two named arguments. `next` is a function which can optionally be called to run the core behavior. `context` contains extra information which may be useful.
```

Comment:
```suggestion
The `behaviorCallback` is passed two named arguments. `next` is a function which can optionally be called to run the old behavior. `context` contains extra information which may be useful.
```

I am not sure if `old` is the best word here, but I'd replace core for something else because if there are other transformers registered there is no guarantee the `next()` corresponds to the core behavior.